doc/howto/write_a_view/index.md

```diff
@@ -0,0 +1,429 @@
+
+    auto v = vec | my_reverse_complement;
+
+    debug_stream << v << '\n';                      // prints "TAATCTGGT"
+}
+```
+
+The adaptor consists of `std::view::reverse` combined with `std::view::transform`, this time the lambda just
+performs the call to `seqan3::complement`.
+\endsolution
+
+# A full custom view implementation
+
+Using existing adaptors only works to a certain degree, sometimes you will need to implement a full view.
+As we have seen above, it is easy to implement a view that presents the complement of a nucleotide range just using
+existing adaptors.
+We will still use this as an example for a full implementation so we will create a non-generic transform view in
+the next steps.
+
+A full view implementation typically consists of the following components:
+
+  1. The view class template that takes the underlying range as a template parameter.
+     * The class should be derived from `std::ranges::view_interface` which will take care of a lot of boilerplate for
+       us.
+     * The actual functionality of the view is implemented in its iterator and sentinel.
+  2. The type definition of the *adaptor object* which provides the operators.
+  3. The *adaptor object* which is just an instance of the type defined in 2.
+
+## The view class template
+
+A good thing to start with is to think about the iterator and the sentinel of your view.
+The iterator and/or its relation to the sentinel is the way we implement the behaviour that is specific to
+our view.
+We will start with implementing the iterator seperately and later integrate it into the view.
+
+### Iterator
+
+Since we know that in our current usecase we have exactly one element in our view for every element in the
+underlying range, we don't need to change the relation between iterator and sentinel, i.e. "begin == end" on
+our view iff "begin == end" on the underlying range.
+This indicates that we can use the sentinel of the underlying range as-is.
+
+\assignment{Exercise 3: Your first iterator}
+
+Have a peak at the \ref tutorial_concept tutorial again and study the `std::ForwardIterator` concept thoroughly.
+You will now have to implement your own forward iterator.
+
+\snippet doc/howto/write_a_view/solution_it.cpp start
+
+```cpp
+    // YOUR IMPLEMENTATION GOES HERE (OPERATORS, MEMBER TYPES...)
+```
+\snippet doc/howto/write_a_view/solution_it.cpp end
+
+It should wrap the iterator of the underlying range or inherit from it (shown above); in the end, you should be able
+to iterate over the underlying range and print elements like you would with the original iterator, i.e. your iterator
+shall behave exactly as the original in this regard (no transformation, yet).
+
+Some things to keep in mind for the implementation:
+  * When defining a type template (like above), member types are not inherited implicitly (member functions are).
+  * Another reason "just inheriting" is not sufficient, is that some inherited functions return objects or references
```

Comment:
```suggestion
  * Another reason "just inheriting" is not sufficient is that some inherited functions return objects or references
```

Reply:
hm, I am not sure about this one

Reply:
mhm, I'd remove it.

Another reason [...] is that blablabla
Another reason [...], is that blablabla

But I guess people will get the message either way